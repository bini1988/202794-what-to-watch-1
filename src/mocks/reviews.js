
export default [
  {
    id: 1,
    author: `Kate Muir`,
    rating: 8.9,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  }, {
    id: 2,
    author: `Mollie`,
    rating: 3,
    text: `It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.`,
    datetime: `2019-06-14T06:39:07.535Z`,
    date: `2019-06-14T06:39:07.535Z`
  }, {
    id: 3,
    author: `Mollie`,
    rating: 4.5,
    text: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    datetime: `2019-05-27T06:39:07.535Z`,
    date: `2019-05-27T06:39:07.535Z`
  }, {
    id: 4,
    author: `Mollie`,
    rating: 2,
    text: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    datetime: `2019-05-27T06:39:07.535Z`,
    date: `2019-05-27T06:39:07.535Z`
  }, {
    id: 5,
    author: `Mollie`,
    rating: 3,
    text: `It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.`,
    datetime: `2019-06-14T06:39:07.535Z`,
    date: `2019-06-14T06:39:07.535Z`
  }, {
    id: 6,
    author: `Mollie`,
    rating: 5.0,
    text: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    datetime: `2019-05-27T06:39:07.535Z`,
    date: `2019-05-27T06:39:07.535Z`
  }, {
    id: 7,
    author: `Mollie`,
    rating: 2.5,
    text: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    datetime: `2019-05-27T06:39:07.535Z`,
    date: `2019-05-27T06:39:07.535Z`
  }
];
