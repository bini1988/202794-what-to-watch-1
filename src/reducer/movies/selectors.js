import {createSelector} from "reselect";
import NameSpace from "../name-spaces";

export const ALL_GENRES_GROUP = `All genres`;

/**
 * Сгруппировать массив фильмов по жанрам
 * @param {Object[]} movies Массив элементов
 * @return {Object}
 */
const groupByGenres = (movies) => {
  const genresGroups = {[ALL_GENRES_GROUP]: movies};

  return movies.reduce((out, it) => {
    out[it.genre] = out[it.genre] || [];
    out[it.genre].push(it);

    return out;
  }, genresGroups);
};

/**
 * Получение списка фильмов
 * @param {Object} state Текущее состояние redux стора
 * @return {Object[]}
 */
export const getMovies = (state) => {
  return state[NameSpace.Movies].items;
};

/**
 * Получение фильма по id
 * @param {Object} state Текущее состояние redux стора
 * @param {string} id Id фильма
 * @return {Object[]}
 */
export const getMovieById = (state, id) => {
  return state[NameSpace.Movies].items.find((it = {}) => it.id === +id);
};

/**
 * Получить текущий промо фильм
 * @param {Object} state Текущее состояние redux стора
 * @return {Object}
 */
export const getPromoMovie = (state) => {
  return state[NameSpace.Movies].promoItem;
};

/**
 * Получить список фильмов добавленных в список «к просмотру»
 * @param {Object} state Текущее состояние redux стора
 * @return {Object}
 */
export const getMyListMovies = (state) => {
  return state[NameSpace.Movies].myListItems;
};

/**
 * Получение активного жанра
 * @param {Object} state Текущее состояние redux стора
 * @return {Object[]}
 */
export const getActiveGenre = (state) => {
  return state[NameSpace.Movies].activeGenre;
};

/**
 * Получение списка фильмов сгруппированног по жанрам
 * @param {Object} state Текущее состояние redux стора
 * @return {Object}
 */
export const getMoviesByGenres = createSelector(getMovies, groupByGenres);

/**
 * Получение списка фильмов заданного жанра
 * @param {Object} state Текущее состояние redux стора
 * @param {string} genre Жанр фильма
 * @return {Object[]}
 */
export const getMoviesByGenre = (state, genre) => {
  return getMoviesByGenres(state)[genre];
};

/**
 * Получение списка жанров
 * @param {Object} state Текущее состояние redux стора
 * @return {string[]}
 */
export const getMoviesGenres = (state) => {
  const MAX_VISIBLE_GENRES = 10;
  const genres = getMoviesByGenres(state);
  return Object.keys(genres).slice(0, MAX_VISIBLE_GENRES);
};
