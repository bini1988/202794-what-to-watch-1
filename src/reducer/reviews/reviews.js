import {NotificationManager} from "react-notifications";
import {normalizeItems} from "../../utils";

export const ActionTypes = {
  STORE_REVIEWS: `STORE_REVIEWS`,
};

export const ActionCreator = {
  /**
   * Сохранить список отзывов к фильму
   * @param {number} id ID фильма
   * @param {Object} items Map-объект отзывов
   * @param {string[]} itemsIds Список id's отзывов
   * @return {Object}
   */
  storeReviews: (id, items, itemsIds) => {
    return {type: ActionTypes.STORE_REVIEWS, meta: {id, itemsIds}, payload: items};
  },
};

export const Operation = {
  /**
   * Получить список отзывов к фильму
   * @param {number} id ID фильма
   * @return {Promise}
   */
  fetchMovieReviews: (id) => {
    return (dispath, getState, api) => {
      return api.fetchMovieReviews(id)
        .then((reviews) => {
          const {items, itemsIds} = normalizeItems(reviews);
          dispath(ActionCreator.storeReviews(id, items, itemsIds));
        }).catch((error) => {
          NotificationManager.error(error.message);
        });
    };
  },
  /**
   * Опубликовать отзыв к фильму
   * @param {number} id ID фильма
   * @param {Object} review Отзыв к фильму
   * @param {number} review.rating Оценка фильма
   * @param {string} review.comment Текст отзыва
   * @return {Promise}
   */
  submitMovieReview: (id, review) => {
    return (dispath, getState, api) => {
      return api.submitMovieReview(id, review)
        .then((reviews) => {
          const {items, itemsIds} = normalizeItems(reviews);
          dispath(ActionCreator.storeReviews(id, items, itemsIds));
        }).catch((error) => {
          NotificationManager.error(error.message);
          throw error;
        });
    };
  },
};


export const initialState = {
  /** Map-объект фильм -> массив отзывов */
  movieToItems: {},
  /** Map-объект отзывов */
  items: {},
};

export default (state = initialState, action) => {
  switch (action.type) {
    case ActionTypes.STORE_REVIEWS:
      return {
        ...state,
        movieToItems: {
          ...state.movieToItems,
          [action.meta.id]: action.meta.itemsIds,
        },
        items: {
          ...state.items,
          ...action.payload,
        },
      };
    default:
      return state;
  }
};
