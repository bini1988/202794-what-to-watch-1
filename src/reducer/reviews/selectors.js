import NameSpace from "../name-spaces";

/**
 * Получение таблицы отзывов к фильмам
 * @param {Object} state Текущее состояние redux стора
 * @return {Object[]}
 */
export const getReviews = (state) => {
  return state[NameSpace.Reviews].items;
};
