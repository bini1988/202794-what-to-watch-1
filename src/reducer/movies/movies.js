import {normalizeItems, groupItemsBy} from "../../utils";
import {getMovieById} from "./selectors";

export const ActionTypes = {
  CHANGE_ACTIVE_GENRE: `CHANGE_ACTIVE_GENRE`,
  STORE_MOVIES: `STORE_MOVIES`,
  STORE_MY_LIST_MOVIES: `STORE_MY_LIST_MOVIES`,
  STORE_MOVIE: `STORE_MOVIE`,
};

export const ActionCreator = {
  /**
   * Изменить фильтр списка фильмов по жанру
   * @param {string} genre Жанр фильма
   * @return {Object}
   */
  changeActiveGenre: (genre) => {
    return {type: ActionTypes.CHANGE_ACTIVE_GENRE, payload: genre};
  },
  /**
   * Сохранить список фильмов
   * @param {Object[]} movies Список фильмов
   * @return {Object}
   */
  storeMovies: (movies) => {
    const {items, itemsIds} = normalizeItems(movies);
    const itemsByGenre = groupItemsBy(movies, `genre`, ({id}) => id);

    return {
      type: ActionTypes.STORE_MOVIES,
      meta: {itemsIds, itemsByGenre},
      payload: items
    };
  },
  /**
   * Сохранить фильм
   * @param {Object} movie Фильм
   * @param {Object} [isPromo] Является ли фильм промо
   * @return {Object}
   */
  storeMovie: (movie, isPromo = false) => {
    const id = movie && movie.id;
    return {type: ActionTypes.STORE_MOVIE, meta: {id, isPromo}, payload: movie};
  },
  /**
   * Сохранить список фильмов «к просмотру»
   * @param {Object[]} movies Список фильмов
   * @return {Object}
   */
  storeMyListMovies: (movies) => {
    const {items, itemsIds} = normalizeItems(movies);
    return {
      type: ActionTypes.STORE_MY_LIST_MOVIES,
      meta: {itemsIds},
      payload: items
    };
  },
};

export const Operation = {
  /**
   * Получить список фильмов
   * @return {Promise}
   */
  fetchMovies: () => {
    return (dispath, getState, api) => {
      return api.fetchMovies().then((movies) => {
        dispath(ActionCreator.storeMovies(movies));
      });
    };
  },
  /**
   * Получить фильм с указанным id
   * @param {number} id Индентификатор фильма
   * @return {Promise}
   */
  fetchMovie: (id) => {
    return (dispath, getState) => {
      const state = getState();
      const movie = getMovieById(state, id);

      if (!movie) {
        return dispath(Operation.fetchMovies());
      }

      return Promise.resolve(movie);
    };
  },
  /**
   * Получить текущий промо фильм
   * @return {Promise}
   */
  fetchPromoMovie: () => {
    return (dispath, getState, api) => {
      return api.fetchPromoMovie().then((movie) => {
        return dispath(ActionCreator.storeMovie(movie, true));
      });
    };
  },
  /**
   * Получить список фильмов добавленных в список «к просмотру»
   * @return {MovieCard[]}
   */
  fetchMyListMovies: () => {
    return (dispath, getState, api) => {
      return api.fetchMyListMovies().then((movies) => {
        return dispath(ActionCreator.storeMyListMovies(movies));
      });
    };
  },
  /**
   * Добавить фильм в список «к просмотру»
   * @param {number} id ID фильма
   * @return {MovieCard}
   */
  addMovieToMyList: (id) => {
    return (dispath, getState, api) => {
      return api.addMovieToMyList(id).then((movie) => {
        return dispath(ActionCreator.storeMovie(movie));
      });
    };
  },
};

export const initialState = {
  /** id промо фильма */
  promoMovieId: null,
  /** Массив фильмов из списка «к просмотру» */
  myListMoviesIds: [],
  /** Массив id's cписка фильмов */
  itemsIds: [],
  /** Map-объект фильмов */
  items: {},
};

export default (state = initialState, action = {}) => {
  const {type, meta, payload} = action;

  switch (type) {
    case ActionTypes.CHANGE_ACTIVE_GENRE:
      return {...state, activeGenre: payload};
    case ActionTypes.STORE_MOVIES:
      return {
        ...state,
        itemsIds: meta.itemsIds,
        items: {...state.items, ...payload},
      };
    case ActionTypes.STORE_MOVIE:
      return {
        ...state,
        promoMovieId: meta.isPromo
          ? meta.id : state.promoMovieId,
        items: {...state.items, [meta.id]: payload},
      };
    case ActionTypes.STORE_MY_LIST_MOVIES:
      return {
        ...state,
        myListMoviesIds: meta.itemsIds,
        items: {...state.items, ...payload},
      };
    default:
      return state;
  }
};
