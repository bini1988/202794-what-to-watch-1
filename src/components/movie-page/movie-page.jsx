import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {MovieReviewPropTypes, MovieCardPropTypes} from "../../prop-types";
import {Operation as MoviesOperation} from "../../reducer/movies/movies";
import {getMovieById} from "../../reducer/movies/selectors";
import {getMoviesGenreLike} from "../../reducer/genres/selectors";
import {Operation as ReviewsOperation} from "../../reducer/reviews/reviews";
import {getMovieReviewsById} from "../../reducer/reviews/selectors";

import MovieCard from "../movie-card/movie-card";
import PageHeader from "../page-header/page-header";
import PageTitle from "../page-title/page-title";
import PageFooter from "../page-footer/page-footer";
import UserBlock from "../user-block/user-block";
import MoviesCatalog from "../movies-catalog/movies-catalog";

class MoviePage extends PureComponent {
  render() {
    const {movie, movies, reviews, location = {}, onToMyListToggle} = this.props;
    const {hash = ``} = location;
    const tab = hash.slice(1) || undefined;

    return (
      <React.Fragment>
        <MovieCard
          full
          card={movie}
          onToMyListToggle={onToMyListToggle}>
          <MovieCard.Wrapper main>
            <MovieCard.Header component={PageHeader}>
              <PageTitle hidden>{`WTW`}</PageTitle>
              <UserBlock/>
            </MovieCard.Header>
            <MovieCard.Wrapper>
              <MovieCard.Description>
                <MovieCard.PlayButton/>
                <MovieCard.ListButton/>
                <MovieCard.ReviewButton/>
              </MovieCard.Description>
            </MovieCard.Wrapper>
          </MovieCard.Wrapper>
          <MovieCard.InfoWrapper translate>
            <MovieCard.Poster mode="big"/>
            <MovieCard.About
              reviews={reviews}
              tab={tab}/>
          </MovieCard.InfoWrapper>
        </MovieCard>
        <div className="page-content">
          <MoviesCatalog
            likeThis
            title="More like this"
            movies={movies}/>
          <PageFooter/>
        </div>
      </React.Fragment>
    );
  }

  componentDidMount() {
    const {match} = this.props;
    const id = match && match.params.id;

    this.props.onMovieFetch(id);
    this.props.onMovieReviewsFetch(id);
  }
}

MoviePage.propTypes = {
  /** Карточка фильма */
  movie: MovieCardPropTypes,
  /** Список похожих фильмов */
  movies: MoviesCatalog.propTypes.movies,
  /** Отзывы к фильму */
  reviews: PropTypes.arrayOf(
      MovieReviewPropTypes,
  ),
  /** React Router location */
  location: PropTypes.object,
  /** React Router match */
  match: PropTypes.object,
  /** Получить фильм */
  onMovieFetch: PropTypes.func,
  /** Получить список отзывов к фильму */
  onMovieReviewsFetch: PropTypes.func,
  /** Добавить/удалить фильм из списока «к просмотру» */
  onToMyListToggle: PropTypes.func,
  /** Вложенные элементы */
  children: PropTypes.any,
};

const mapStateToProps = (state, {match}) => {
  const id = match && match.params.id;
  const movie = getMovieById(state, id);
  const reviews = getMovieReviewsById(state, id);
  const movies = getMoviesGenreLike(state, id);

  return {movie, reviews, movies};
};
const mapDispatchToProps = {
  onMovieFetch: MoviesOperation.fetchMovie,
  onMovieReviewsFetch: ReviewsOperation.fetchMovieReviews,
  onToMyListToggle: MoviesOperation.toggleMovieToMyList,
};

export {MoviePage};
export default connect(mapStateToProps, mapDispatchToProps)(MoviePage);

