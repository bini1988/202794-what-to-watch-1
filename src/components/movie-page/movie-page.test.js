import React from "react";
import {MemoryRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import {MoviePageView} from "./movie-page";
import movies from "../../mocks/movies";
import card from "../../mocks/movie-card";
import reviews from "../../mocks/reviews";

const initialState = {};
const mockStore = configureStore();

it(`MoviePageView correctly renders default markup`, () => {
  const tree = renderer.create(
      <Provider store={mockStore(initialState)}>
        <MemoryRouter
          initialEntries = {[`/`]}>
          <MoviePageView
            movie={card}
            movies={movies}
            reviews={reviews}
            onMovieFetch={() => {}}
            onMovieReviewsFetch={() => {}}
            onToMyListToggle={() => {}}
          />
        </MemoryRouter>
      </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
