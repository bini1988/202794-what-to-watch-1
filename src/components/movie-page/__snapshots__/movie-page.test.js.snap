// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePageView correctly renders default markup 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "background": "#9B7E61",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="movie-card__bg"
        >
          <img
            alt="War of the Worlds"
            src="https://es31-server.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg"
          />
        </div>
        <h1
          className="page-title visually-hidden"
        >
          WTW
        </h1>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            War of the Worlds
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Adventure
            </span>
            <span
              className="movie-card__year"
            >
              2005
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/film/1/player"
              onClick={[Function]}
            >
              <svg
                height={19}
                viewBox="0 0 19 19"
                width={19}
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height={20}
                viewBox="0 0 19 20"
                width={19}
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/film/1/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="War of the Worlds"
            height="327"
            src="https://es31-server.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#Overview"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#Details"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#Reviews"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              6.5
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Good
              </span>
              <span
                className="movie-rating__count"
              >
                386834 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            As Earth is invaded by alien tripod fighting machines, one family fights for survival.
            <p
              className="movie-card__director"
            >
              <strong>
                Director: Steven Spielberg
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: Tom Cruise, Dakota Fanning, Tim Robbin and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          id="movie-1"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/film/1"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height={175}
                muted={true}
                onCanPlayThrough={[Function]}
                onEnded={[Function]}
                onLoadStart={[Function]}
                onPause={[Function]}
                onTimeUpdate={[Function]}
                poster="https://es31-server.appspot.com/wtw/static/film/preview/midnight-special.jpg"
                preload="none"
                src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
                width={280}
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              Midnight Special
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id="movie-2"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/film/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height={175}
                muted={true}
                onCanPlayThrough={[Function]}
                onEnded={[Function]}
                onLoadStart={[Function]}
                onPause={[Function]}
                onTimeUpdate={[Function]}
                poster="https://es31-server.appspot.com/wtw/static/film/preview/pulp-fiction.jpg"
                preload="none"
                src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
                width={280}
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              Pulp Fiction
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id="movie-3"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/film/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height={175}
                muted={true}
                onCanPlayThrough={[Function]}
                onEnded={[Function]}
                onLoadStart={[Function]}
                onPause={[Function]}
                onTimeUpdate={[Function]}
                poster="https://es31-server.appspot.com/wtw/static/film/preview/bronson.jpg"
                preload="none"
                src="http://media.xiph.org/mango/tears_of_steel_1080p.webm"
                width={280}
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              Bronson
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id="movie-4"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/film/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height={175}
                muted={true}
                onCanPlayThrough={[Function]}
                onEnded={[Function]}
                onLoadStart={[Function]}
                onPause={[Function]}
                onTimeUpdate={[Function]}
                poster="https://es31-server.appspot.com/wtw/static/film/preview/what-we-do-in-the-shadows.jpg"
                preload="none"
                src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
                width={280}
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              What We Do in the Shadows
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id="movie-5"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/film/5"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height={175}
                muted={true}
                onCanPlayThrough={[Function]}
                onEnded={[Function]}
                onLoadStart={[Function]}
                onPause={[Function]}
                onTimeUpdate={[Function]}
                poster="https://es31-server.appspot.com/wtw/static/film/preview/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                preload="none"
                src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
                width={280}
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </h3>
          </a>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
